package.json

```diff
@@ -18,10 +18,8 @@
   },
   "homepage": "https://github.com/apache/arrow-site#readme",
   "dependencies": {
-    "bootstrap": "^4.6.0",
-    "jquery": "^3.5.1",
-    "popper.js": "^1.16.1",
-    "webpack": "^5.76.0",
+    "bootstrap": "^5.3.2",
```

Comment:
Updating Bootstrap 4 -> 5 may break design. Could you check it?

Reply:
It seems ok. Most of the documentation is built in the other Arrow repositories.

Reply:
Could you share what you checked?
This PR also updates `jekyll-feed`. So we need to check at least https://arrow.apache.org/feed.xml too.

Reply:
Providing preview is also helpful.
See also: https://github.com/apache/arrow-site#forks

Reply:
The current preview workflow at https://github.com/apache/arrow-site/blob/main/.github/workflows/deploy.yml fails with newer Jekyll as the one used by GitHub is older.

The current site can be previewed at https://bkmgit.github.io/arrow-site - without changes

Associated workflow is at:
https://github.com/bkmgit/arrow-site/blob/bkmgit-patch-1/.github/workflows/preview.yml
Can make a pull request to add this so that creating preview sites is easier.
- It does not download the crossbow artifacts, though can add these if they are helpful
- One would need to change permissions to enable deployment, see https://stackoverflow.com/questions/76937061/branch-master-is-not-allowed-to-deploy-to-github-pages-due-to-environment-prot
- If the preview site is created within a subfolder of https://apache.github.io/arrow-site/ say https://apache.github.io/arrow-site/preview/2024-01-06-12-30 then a cron job can be used to remove the preview but having the preview hosted from the origin of the pull request seems sufficient.  Just need to document what to do.

Reply:
> The current site can be previewed at https://bkmgit.github.io/arrow-site - without changes

We want a preview "with" these changes for this PR.

> Associated workflow is at:
> https://github.com/bkmgit/arrow-site/blob/bkmgit-patch-1/.github/workflows/preview.yml
> Can make a pull request to add this so that creating preview sites is easier.

Can we extend our existing `deploy.yml` workflow instead of creating a new workflow?
It already has some codes for preview such as https://github.com/apache/arrow-site/blob/main/.github/workflows/deploy.yml#L63 .

> If the preview site is created within a subfolder of https://apache.github.io/arrow-site/

We want to create a preview in a fork repository such as https://github.com/bkmgit/arrow-site/ not apache/arrow-site.

Reply:
Modified existing `deploy.yml`. 
- https://github.com/apache/arrow-site/pull/453 uses baseurl
From https://mademistakes.com/mastering-jekyll/site-url-baseurl/ probably using baseurl is good. Will come back to this once working preview is merged.

Reply:
Preview is available at:
https://bkmgit.github.io/arrow-site/

Rss feed validation gives a few errors, but these are also present on current rss feed:
https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fbkmgit.github.io%2Farrow-site%2Ffeed.xm

> 
>    Recommendations
> 
> This feed is valid, but interoperability with the widest range of feed readers could be improved by implementing the following recommendations.
> 
>     [line 60](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fbkmgit.github.io%2Farrow-site%2Ffeed.xml#l60), column 1859: Two entries with the same value for atom:updated: 2023-11-09T00:00:00-05:00 [[help](https://validator.w3.org/feed/docs/warning/DuplicateUpdated.html)]
> 
>         ... ished><updated>2023-11-09T00:00:00-05:00</updated><id>https://bkmgit.git ...
>                                                      ^
> 
>     [line 1090](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fbkmgit.github.io%2Farrow-site%2Ffeed.xml#l1090), column 0: Invalid HTML: Unexpected '-' after '--' found in comment. [[help](https://validator.w3.org/feed/docs/warning/NotHtml.html)]
> 
>         implementation, see the latest <a href="https://github.com/apache/arrow-rs/t ...

Reply:
The former isn't problem because we have 2 posts on 2023-11-09.

The latter will be fixed by the following. Could you send a separated PR?

```diff
diff --git a/_posts/2023-08-05-datafusion_fast_grouping.md b/_posts/2023-08-05-datafusion_fast_grouping.md
index f8b5a50e5ef..57fc1b70d87 100644
--- a/_posts/2023-08-05-datafusion_fast_grouping.md
+++ b/_posts/2023-08-05-datafusion_fast_grouping.md
@@ -25,7 +25,7 @@ limitations under the License.
 {% endcomment %}
 -->
 
-<!--- Converted from Google Docs using https://www.buymeacoffee.com/docstomarkdown --->
+<!-- Converted from Google Docs using https://www.buymeacoffee.com/docstomarkdown -->
 
 ## Aggregating Millions of Groups Fast in Apache Arrow DataFusion
 
```

Reply:
The preview site has `<meta name="generator" content="Jekyll v4.2.0" />`.
Is it really generated with Jekyll 4.3.3?

Reply:
PR to fix rss https://github.com/apache/arrow-site/pull/461

Thanks, had served pages from incorrect directory. Needed to add base_url to config to
get correct urls in the preview.  Using environment variables seems to repeat the base_url
so have put this in the extra_config.yml file

Preview at https://bkmgit.github.io/arrow-site/

Reply:
Need to check navbar, other things seem to work.

Reply:
@kou Dropdowns work now. Let me know if anything else is needed.